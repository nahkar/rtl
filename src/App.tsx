import React, { useEffect, useState } from 'react';
import logo from './logo.svg';
import './App.css';

interface SearchProps{
  value: string;
  onChange: (value:string) => void
}

interface User{
  id:  number;
  name: string;
}

const getUser = () => Promise.resolve({id: 1, name: 'User'});

export function Search({value, onChange}:SearchProps) {
  return (
		<div>
			<label htmlFor="search">Search : </label>
			<input id="search" type="text" value={value} onChange={(e) => onChange(e.target.value)} />
		</div>
	);
}


function App() {
  const [user, setUser] = useState<User>(); 
  const [search, setSearch] = useState('')

  useEffect(() => {
    const loadUser = async () => {
      const userData = await getUser();
      setUser(userData);
    };
    loadUser();
  }, []);
  return (
		<div className="App">
			<header className="App-header">
				{user && <h1 data-testid="search">{user.name}</h1>}
				<Search value={search} onChange={setSearch} />
			</header>
		</div>
	);
}

export default App;
