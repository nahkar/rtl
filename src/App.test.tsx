import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App component', () => {
	it('Renders App', async () => {
		render(<App />);
		expect(screen.queryByText('User')).toBeNull();
		screen.debug()
		expect(await screen.findByText('User')).toBeInTheDocument();
    screen.debug();
	});
});
