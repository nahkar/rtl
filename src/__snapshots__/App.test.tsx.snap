// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        React
      </h1>
      <div>
        <label
          for="search"
        >
          Search : 
        </label>
        <input
          id="search"
          type="text"
          value=""
        />
      </div>
    </header>
  </div>
</DocumentFragment>
`;
